### DISCUSSION TITLE
inputs.data undefined after passing through transformer.deserialize

### DISCUSSION THREAD

**Discussion by af4oz** [trpc/trpc#1682] [Q&A] [answered]
I am using superjson transformer.

// from trpc-playground
```js
await mutation('user.signUp',{
  data: {
    password: "123",
    name: "123213",
    email: "tes122t@test.com"
  }
})
```
It throws `invalid type,expected object, received undefined` error.

below are few observations  from console.logging `trpc-server/dist`:
```sh
POST /trpc/user.signUp {
  '0': {
    data: { password: '123', name: '123213', email: 'tes122t@test.com' }
  }
}
{ bodyResult: { ok: true, data: { '0': [Object] } } }
{ rawInput: { '0': { data: [Object] } }, type: 'object' }

//  within `getInputs`  and before `deserializeInputValue`
{
  rawValue: {
    data: { password: '123', name: '123213', email: 'tes122t@test.com' }
  },
  type: 'object'
}

// after `getInputs`
{ inputs: { '0': undefined } }
```

**Comment by KATT** [comment, marked as answer]
Sorry, what is the question?

Please make a reproduction issue.

I find it very unlikely that basic mutations would be broken, they're tested hundreds of times

**Reply by af4oz** [reply]
Iam using superjson transformer for server and trpc-playground to test.Here's an example [setup](https://codesandbox.io/s/broken-violet-mj78y2?file=/src/server.ts) (you can use above mutation query).

If we remove superjson, it works fine.

**Reply by KATT** [reply]
If you use superjson, you have to send a superjson compat object

https://github.com/blitz-js/superjson#basic-usage

Unsure why you're calling trpc routes directly without a trpc client? That's the real question here

**Reply by KATT** [reply]
If you're using a client, you also need to add the transformer there

**Reply by KATT** [reply]
Unsure if the trpc playground supports transformers - @sachinraja?

**Reply by sachinraja** [reply]
It does not unless you're using preact, in which case you could use a component instead of the pre-compiled HTML build. Ideally I could compile for react too, but I haven't gotten around to it. A quick fix I could do would be offering an option to use superjson, I'll try to get around to doing one of the above soon, but I'm very busy atm.

**Reply by sachinraja** [reply]
I've added an option to use the superjson transformer in tRPC Playground @zkindest. You can add it to your config like this:
```ts
app.use(
    playgroundEndpoint,
    await expressHandler({
      trpcApiEndpoint,
      playgroundEndpoint,
      router: appRouter,
      request: {
        superjson: true,
      },
    }),
  )
```

There's a bit of a complex bundling pipeline so it's not really feasible to offer a React component here (`typescript` is a bundled dependency).
